src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/BooleanExpressionComplexityCheck.java

```diff
@@ -29,12 +29,69 @@
 import com.puppycrawl.tools.checkstyle.utils.CheckUtil;
 
 /**
- * Restricts nested boolean operators (&amp;&amp;, ||, &amp;, | and ^) to
- * a specified depth (default = 3).
- * Note: &amp;, | and ^ are not checked if they are part of constructor or
- * method call because they can be applied to non boolean variables and
- * Checkstyle does not know types of methods from different classes.
+ * <p>
+ * Restrict the number of {@code &amp;&amp;}, {@code ||}, {@code &amp;}, {@code |} and
+ * {@code ^} in an expression.
+ * </p>
+ * <p>
+ * Rationale: Too many conditions leads to code that is difficult to read
+ * and hence debug and maintain.
+ * </p>
+ * <p>
+ * Note that the operators {@code &amp;} and {@code |} are not only integer bitwise
+ * operators, they are also the
+ * <a href="https://docs.oracle.com/javase/specs/jls/se11/html/jls-15.html#jls-15.22.2">
+ * non-shortcut versions</a> of the boolean operators. {@code &amp;&amp;} and {@code ||}.
```

Comment:
Why the period between the end of the sentence and just the display of `&& and ||`?

Reply:
done